[GStreamer] Move some DMABuf-related code from the player to a new DMABufUtilities file

philn opened the pull request:
#### ef8296de8bc58c4434fb79656f78a76cea852ca3
<pre>
[GStreamer] Move some DMABuf-related code from the player to a new DMABufUtilities file
<a href="https://bugs.webkit.org/show_bug.cgi?id=276481">https://bugs.webkit.org/show_bug.cgi?id=276481</a>

Reviewed by Xabier Rodriguez-Calvar.

Refactoring aiming to reduce a bit the size of MediaPlayerPrivateGStreamer::pushDMABufToCompositor()
and improve readability.

* Source/WebCore/platform/SourcesGStreamer.txt:
* Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.cpp: Added.
(WebCore::DMABufDestination::DMABufDestination):
(WebCore::DMABufDestination::~DMABufDestination):
(WebCore::DMABufDestination::copyPlaneData):
(WebCore::fillSwapChainBuffer):
* Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.h: Added.
(WebCore::dmaBufFourccValue):
(WebCore::dmaBufColorSpaceForColorimetry):
(WebCore::DMABufDestination::mappingLock):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::pushDMABufToCompositor):
(WebCore::fourccValue): Deleted.
(WebCore::colorSpaceForColorimetry): Deleted.

Canonical link: <a href="https://commits.webkit.org/280906@main">https://commits.webkit.org/280906@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/31ac88193b5ca44e7e783203b46c91afa1bdaa91

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57991 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37319 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10467 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61616 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8436 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60119 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44955 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8624 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46995 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6008 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60021 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34999 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50126 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27826 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31763 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7421 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7440 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53709 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7689 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63304 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1905 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7758 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54220 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1912 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50137 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54359 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12827 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1632 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33148 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34234 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35318 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33979 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.cpp:
```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright (C) 2024 Metrological Group B.V.
```
Please add Igalia's copyright as well.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.h:
```diff
@@ -0,0 +1,109 @@
+/*
+ *  Copyright (C) 2024 Metrological Group B.V.
```
ditto

calvaris commented on Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.h:
```diff
@@ -0,0 +1,109 @@
+
+    void copyPlaneData(GstVideoFrame*, unsigned planeIndex);
+
+    bool valid { false };
```
isValid

calvaris commented on Source/WebCore/platform/network/DataURLDecoder.cpp:
```diff
@@ -31,6 +31,7 @@
 #include <pal/text/DecodeEscapeSequences.h>
 #include <pal/text/TextEncoding.h>
 #include <wtf/MainThread.h>
+#include <wtf/NeverDestroyed.h>
```
Leftover from something else?

philn replied:
No, unified build fix!

philn committed:
[GStreamer] Move some DMABuf-related code from the player to a new DMABufUtilities file
https://bugs.webkit.org/show_bug.cgi?id=276481

Reviewed by Xabier Rodriguez-Calvar.

Refactoring aiming to reduce a bit the size of MediaPlayerPrivateGStreamer::pushDMABufToCompositor()
and improve readability.

* Source/WebCore/platform/SourcesGStreamer.txt:
* Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.cpp: Added.
(WebCore::DMABufDestination::DMABufDestination):
(WebCore::DMABufDestination::~DMABufDestination):
(WebCore::DMABufDestination::copyPlaneData):
(WebCore::fillSwapChainBuffer):
* Source/WebCore/platform/graphics/gstreamer/DMABufUtilities.h: Added.
(WebCore::dmaBufFourccValue):
(WebCore::dmaBufColorSpaceForColorimetry):
(WebCore::DMABufDestination::mappingLock):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::pushDMABufToCompositor):
(WebCore::fourccValue): Deleted.
(WebCore::colorSpaceForColorimetry): Deleted.

Canonical link: https://commits.webkit.org/280906@main
